From a2342b95587f4b242317e8c1bcff3f90b7c9aabd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Mon, 1 Sep 2025 18:56:51 +0200
Subject: [PATCH 885/954] feat(stm32mp21): add CRYP node in STM32MP21 platforms
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This patch add the node of CRYP1 and CRYP2 peripheral that is available
for STM32MP21xF and STM32MP21xC.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git ffed222fd20b277b13ab2324febd208c211de8b3 feat(stm32mp21): add CRYP node in STM32MP21 platforms]
Change-Id: I5e5af80d1e3d563c54869b86704520b07c385131
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/486160
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 fdts/stm32mp21xc.dtsi | 16 ++++++++++++++++
 fdts/stm32mp21xf.dtsi | 16 ++++++++++++++++
 2 files changed, 32 insertions(+)

diff --git a/fdts/stm32mp21xc.dtsi b/fdts/stm32mp21xc.dtsi
index 700d399101..713eb7ae12 100644
--- a/fdts/stm32mp21xc.dtsi
+++ b/fdts/stm32mp21xc.dtsi
@@ -6,6 +6,22 @@
 
 / {
         soc@0 {
+		cryp1: cryp@42030000 {
+			compatible = "st,stm32mp1-cryp";
+			reg = <0x42030000 0x400>;
+			clocks = <&rcc CK_BUS_CRYP1>;
+			resets = <&rcc CRYP1_R>;
+			status = "disabled";
+		};
+
+		cryp2: cryp@42040000 {
+			compatible = "st,stm32mp1-cryp";
+			reg = <0x42040000 0x400>;
+			clocks = <&rcc CK_BUS_CRYP2>;
+			resets = <&rcc CRYP2_R>;
+			status = "disabled";
+		};
+
 		saes: saes@42030c00 {
 			compatible = "st,stm32-saes";
 			reg = <0x42030c00 0x400>;
diff --git a/fdts/stm32mp21xf.dtsi b/fdts/stm32mp21xf.dtsi
index 700d399101..713eb7ae12 100644
--- a/fdts/stm32mp21xf.dtsi
+++ b/fdts/stm32mp21xf.dtsi
@@ -6,6 +6,22 @@
 
 / {
         soc@0 {
+		cryp1: cryp@42030000 {
+			compatible = "st,stm32mp1-cryp";
+			reg = <0x42030000 0x400>;
+			clocks = <&rcc CK_BUS_CRYP1>;
+			resets = <&rcc CRYP1_R>;
+			status = "disabled";
+		};
+
+		cryp2: cryp@42040000 {
+			compatible = "st,stm32mp1-cryp";
+			reg = <0x42040000 0x400>;
+			clocks = <&rcc CK_BUS_CRYP2>;
+			resets = <&rcc CRYP2_R>;
+			status = "disabled";
+		};
+
 		saes: saes@42030c00 {
 			compatible = "st,stm32-saes";
 			reg = <0x42030c00 0x400>;
-- 
2.53.0

